pkg/apis/infrastructure/v1alpha2/awsmachine_types.go

```diff
@@ -0,0 +1,123 @@
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	userdata "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/userdata"
+)
+
+// AWSMachineSpec defines the desired state of AWSMachine
+type AWSMachineSpec struct {
+	// AMI is the reference to the AMI from which to create the machine instance.
+	AMI AWSResourceReference `json:"ami,omitempty"`
+
+	// ImageLookupOrg is the AWS Organization ID to use for image lookup if AMI is not set.
+	ImageLookupOrg string `json:"imageLookupOrg,omitempty"`
+
+	// InstanceType is the type of instance to create. Example: m4.xlarge
+	InstanceType string `json:"instanceType,omitempty"`
+
+	// AdditionalTags is the set of tags to add to an instance, in addition to the ones
+	// added by default by the actuator. These tags are additive. The actuator will ensure
+	// these tags are present, but will not remove any other tags that may exist on the
+	// instance.
+	// +optional
+	AdditionalTags map[string]string `json:"additionalTags,omitempty"`
+
+	// IAMInstanceProfile is a name of an IAM instance profile to assign to the instance
+	// +optional
+	IAMInstanceProfile string `json:"iamInstanceProfile,omitempty"`
+
+	// PublicIP specifies whether the instance should get a public IP.
+	// Precedence for this setting is as follows:
+	// 1. This field if set
+	// 2. Cluster/flavor setting
+	// 3. Subnet default
+	// +optional
+	PublicIP *bool `json:"publicIP,omitempty"`
+
+	// AdditionalSecurityGroups is an array of references to security groups that should be applied to the
+	// instance. These security groups would be set in addition to any security groups defined
+	// at the cluster level or in the actuator.
+	// +optional
+	AdditionalSecurityGroups []AWSResourceReference `json:"additionalSecurityGroups,omitempty"`
+
+	// AvailabilityZone is references the AWS availability zone to use for this instance.
+	// If multiple subnets are matched for the availability zone, the first one return is picked.
+	// +optional
+	AvailabilityZone *string `json:"availabilityZone,omitempty"`
+
+	// Subnet is a reference to the subnet to use for this instance. If not specified,
+	// the cluster subnet will be used.
+	// +optional
+	Subnet *AWSResourceReference `json:"subnet,omitempty"`
+
+	// KeyName is the name of the SSH key to install on the instance.
+	// +optional
+	KeyName string `json:"keyName,omitempty"`
+
+	// RootDeviceSize is the size of the root volume.
+	// +optional
+	RootDeviceSize int64 `json:"rootDeviceSize,omitempty"`
+
+	// AdditionalUserDataFiles specifies extra files to be passed to user_data upon creation.
+	// +optional
+	AdditionalUserDataFiles []userdata.Files `json:"additionalUserDataFiles,omitempty"`
+}
+
+// AWSMachineStatus defines the observed state of AWSMachine
+type AWSMachineStatus struct {
+	// InstanceID is the instance ID of the machine created in AWS
+	// +optional
+	InstanceID *string `json:"instanceID,omitempty"`
```

Comment:
Will it be possible to redetermine the InstanceID from just the spec?

Reply:
I'm not sure what you mean, I copied this from the providerStatus